server/src/main/java/org/opensearch/tasks/TaskThreadExecutions.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.tasks;
+
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.core.ParseField;
+import org.opensearch.core.xcontent.ConstructingObjectParser;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import static org.opensearch.core.xcontent.ConstructingObjectParser.constructorArg;
+
+/**
+ * Task thread execution information.
+ * Writeable TaskThreadExecutions objects are used to represent thread related resource usage of running tasks.
+ * Currently, this only tracks the number of thread executions for a task but can be extended to include additional details as needed.
+ *
+ * @opensearch.internal
+ */
+public class TaskThreadExecutions extends AbstractTaskResourceUsage {
```

Comment:
```suggestion
public class TaskThreadUsage extends AbstractTaskResourceUsage {
```
(to be consistent with TaskResourceUsage